lib/rust/ensogl/core/src/display/world/stats.rs

```diff
@@ -41,25 +46,29 @@ impl {
             monitor.add( debug::monitor::SpriteSystemCount  :: new(&stats) ),
             monitor.add( debug::monitor::SpriteCount        :: new(&stats) ),
         ];
-        Self {stats,monitor,panels}
+        Self {stats,performance,monitor,panels}
     }
 
     /// Start measuring data.
     pub fn begin(&mut self) {
-        if self.visible() {
-            for panel in &self.panels {
-                panel.begin();
-            }
+        if self.visible() || profiling::ENABLED {
+            let time = self.performance.now();
+            self.stats.begin_frame(time);
         }
     }
 
     /// Finish measuring data.
     pub fn end(&mut self) {
-        if self.visible() {
-            for panel in &self.panels {
-                panel.end();
+        if self.visible() || profiling::ENABLED {
+            let time = self.performance.now();
+            self.stats.end_frame(time);
+            if self.visible() {
+                for panel in &self.panels {
+                    panel.end();
+                }
+                self.monitor.draw();
             }
-            self.monitor.draw();
+            profiling::frame_stats::intervals::push_stats(&self.stats.data());
```

Comment:
This looks like a super internal function usage – EnsoGL is pushing some data into a vector stored by another library. This will be fixed after other comments are applied, but this line alone looks like the responsibilities are badly mixed.